### ISSUE TITLE
Change the import name from MySQLdb to mysqlclient to avoid conflicts, or taking over "MySQL-python" on PyPI

### ISSUE THREAD

**Issue by zzzeek** [PyMySQL/mysqlclient-python#44]
It is destabilizing to expect that a particular application that wishes to make use of "mysqlclient" under Python 2K must therefore in one step use mysqlclient for _all_ applications that share the same virtual or actual Python environment.    it should be be possible for some applications to use the traditional MySQLdb and for others to use mysqlclient.

Other DBAPIs that began as forks or ports of others have followed this practice, e.g. fdb is a port of kinterbasdb, psycopg2cffi is a pypy port of psycopg2, etc.   It only creates confusion and reduces options when two totally different codebases share the exact same module name.   Especially on Linux distributions it means that two packages will necessarily overwrite each other and be in conflict.

**Comment by methane** [comment]
There are many products depending MySQL-python. And MySQL-python can't be compiled on some new distributions.
In such cases, `pip install mysqlclient` helps people want run application without modifying it.
It is like distribute and pillow overwrite `setuptools` and `PIL` package.

**Comment by zzzeek** [comment]
> And MySQL-python can't be compiled on some new distributions.

so what?  those distributions should only distribute mysqlclient.

> In such cases, pip install mysqlclient helps people want run application without modifying it.

what application *hardcodes* the name MySQLdb inside of it, and if so, how is that not a bug on their end?  

> It is like distribute and pillow overwrite setuptools and PIL package.

distribute's overwrite of the "setuptools" name was a disaster and they had to undo it, once setuptools became maintained again.    PIL has been a totally dead project for much much longer than MySQLdb.

**Comment by zzzeek** [comment]
also, one thing this name collision does is make it very difficult and inconvenient for me to test SQLAlchemy against both packages.    It means that for me to test one or the other, I need to use entirely separate virtual environments.  Dropping support for mysql-python from SQLAlchemy is not an option.   It is also not safe to assume these two libraries have identical behavior; that's actually impossible, as MySQL-Python has bugs that this library promises to fix.   SQLAlchemy has behaviors and workarounds that may no longer apply to mysqlclient, or may need behaviors and workarounds that are specific to mysqlclient.   The testing burden for downstream libraries here increases.

**Comment by zzzeek** [comment]
@methane  - have you tried contacting @farcepest to request being given access to take over https://pypi.python.org/pypi/MySQL-python/ on Pypi ?

**Comment by methane** [comment]
> @methane - have you tried contacting @farcepest to request being given access to take over https://pypi.python.org/pypi/MySQL-python/ on Pypi ?

@rbu sent mail about it on 2014/12.

I have not sent such mail yet since I'm not good at writing English to write modest mail.

**Comment by methane** [comment]
> so what? those distributions should only distribute mysqlclient.

How people run Trac with MySQL without rewriting imports?

> > In such cases, pip install mysqlclient helps people want run application without modifying it.

> what application hardcodes the name MySQLdb inside of it, and if so, how is that not a bug on their end?

Trac.  And Django only support MySQL-python officially.
PeeWee, SQLAlchemy and other ORMs should support new package name too.

> >It is like distribute and pillow overwrite setuptools and PIL package.

> distribute's overwrite of the "setuptools" name was a disaster

I think there are more disaster if distribute use other name.  People may be required to rewrite `import setuptools` in `setup.py` to install software on platform setuptools doesn't support.
Using different package name for drop-in replacement of unmaintained library requires efforts to many people (including not Python programmer).

> and they had to undo it, once setuptools became maintained again.

I stop this project happily when MySQL-python become maintained again and support PY3 well.

**Comment by methane** [comment]
> also, one thing this name collision does is make it very difficult and inconvenient for me to test SQLAlchemy against both packages. It means that for me to test one or the other, I need to use entirely separate virtual environments.

I don't know how difficult to test both with tox.

>  It is also not safe to assume these two libraries have identical behavior; that's actually impossible, as MySQL-Python has bugs that this library promises to fix. SQLAlchemy has behaviors and workarounds that may no longer apply to mysqlclient, or may need behaviors and workarounds that are specific to mysqlclient.

I see. mysqlclient should provide way to distinguish from MySQL-python even if using same package name.

**Comment by methane** [comment]
PyMySQL has `install_as_MySQLdb()` to replace MySQL-python.
But it's inconvenient since adding `import pymysql; pymysql.install_as_MySQLdb()` in right place is difficult.

So I don't want to remove "pip installable drop-in replacement for MySQL-python" feature.

I'm thinking about providing "mysqldb-mysqlclient" package which provide `MySQLdb` package. It only do `import mysqlclient; mysqlclient.install_as_MySQLdb()`.
When people want separated package, just "pip install mysqlclient".
When people want drop-in replacement, "pip install mysqldb-mysqlclient".

Another way is creating another fork. It fixes only build&install issue and keeps behavior including bug.

**Comment by zzzeek** [comment]
Well Trac would need to support dynamic database DBAPI names.    I'd consider it a bug in Trac.  They support multiple backends like SQLite and Postgresql so it isn't like they are using the special non-DBAPI features of MySQLdb.

Providing an alternate package is another approach.  But mostly, I'd like @farcepest to let you just take over the Pypi entry.  If he's unresponsive on this, then it seems most people are already preferring that mysqlclient just walk over the MySQLdb import name even though I'd prefer we didn't.

**Comment by zzzeek** [comment]
Also, if you want to reach out to @farcepest and are concerned about English, send me what you want to send and I can suggest edits.

**Comment by methane** [comment]
My most desirable future is merging mysqlclient into MySQL-python.
This is why I didn't change package name.

But I want to help Debian, Ubuntu, Redhat, CentOS and OpenStack moving to Python 3.
Especially, I hope Ubuntu 16.04 LTS and Debian Jessie are completely moved to Python 3.
If current status of MySQL-python and mysqlclient makes difficult to moving Python 3,
I'll give up about merging back and change package name.

First of all, I'll contact to @farcepest and @paultag (https://lists.debian.org/debian-devel-announce/2015/04/msg00005.html ).

**Comment by rbu** [comment]
@zzzeek, I emailed @farcepest five months ago on December 3, 2014 and again 3 months ago on February 3, 2015 simply to request input or give a statement what his intentions for the project are. There was no response.  @lecram and @methane were in CC.

I think the best course of action would be to request a project takeover on PyPI.

@methane, I agree with @zzzeek on the naming issue. It should be possible to maintain the fork with a different module name and have a second package on PyPI that just provides the compatibility drop-in.

**Comment by methane** [comment]
I found bug about this problem.
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=768096

**Comment by thomasgoirand** [comment]
Methane,

FYI, and if you didn't get it, the bug ends with mysqlclient being uploaded as a replacement for python-mysqldb in Debian experimental. In other words, the maintainer of the python-mysqldb package decided to replace mysql-python by mysqlclient.

Given the lack of response from the maintainer of mysql-python, I would suggest to go ahead with the take over of mysql-python in PyPi, though IMO, you should make a last attempt at contacting him again, explaining your intentions. Best would be to do this with CC to a public mailing list, and give him an ultimatum (after X number of weeks without reply, here's what is going to happen...).

**Comment by methane** [comment]
I want to discuss with OpenStack, Debian and RedHat developers to change or not change import name...

**Comment by tiwilliam** [comment]
I think it makes perfect sense to issue a take over request since no life sign of MySQLdb have been shown in over a year, I've tried to contact @farcepest as well without any success. The library have 500k monthly downloads, not being able to patch security issues or bugs is critical. I agree we should send one last mail to @farcepest and CC [debian-python mailing list](https://lists.debian.org/debian-python/).

@methane let me know if you don't have time, and I'll try to push this.

**Comment by tiwilliam** [comment]
I just got contact with Andy and he we'll look into how we can solve the ownership tomorrow.

**Comment by s7v7nislands** [comment]
@tiwilliam any update?

**Comment by tiwilliam** [comment]
Nope, he never came back. Got contact on Google+.

**Comment by rbu** [comment]
http://sourceforge.net/p/pypi/support-requests/ is where the owner of the new package can request a takeover of the existing name

**Comment by brianmay** [comment]
"And Django only support MySQL-python officially." This is wrong. According to https://docs.djangoproject.com/en/1.8/ref/databases/ "mysqlclient is a fork of MySQLdb which notably supports Python 3 and can be used as a drop-in replacement for MySQLdb. At the time of this writing, this is the recommended choice for using MySQL with Django."

**Comment by methane** [comment]
@rbu Thank you for your imformation.
I've posted support request: https://sourceforge.net/p/pypi/support-requests/553/

**Comment by graingert** [comment]
what about mysql-connector-python /cc @geertjanvdk

**Comment by methane** [comment]
If Oracle want to take over MySQL-python on PyPI, and they will upload at least binary wheel for windows and pip installable packeges for unix, I'm +1 for it.

This project is continuous maintenance of MySQL-python 1.2.x.
So I want to release 1.3.x while Oracle release 2.x~, if it's possible.

**Comment by graingert** [comment]
@methane I don't think Oracle have any intentions of uploading anything to PyPI. You should push for PyPI to make the PyMySQL team owners of the names on PyPI.

**Comment by geertjanvdk** [comment]
Currently, the PyPI license gives issues when uploading GPL'ed software. Would be freaking great if Connector/Python is on PyPI, but nothing Oracle can help.. 'MySQL-Python' and 'PyMySQL' are not Oracle's and are not related to Connector/Python.

**Comment by graingert** [comment]
PyPI is 100% compatible with the GPL

**Comment by graingert** [comment]
@geertjanvdk what issue do you see in uploading GPL software to PyPI?

There's even a trove classifier: License :: OSI Approved :: GNU General Public License v2 (GPLv2)

**Comment by geertjanvdk** [comment]
@graingert Nothing to do with listings or classifiers. I should not discuss this, so I will not (is why I shouldn't have mentioned and not replied, my bad) My last [blog post in February](https://geert.vanderkelen.org/2016/cpy-and-pypi), more I can not do, sadly.

**Comment by graingert** [comment]
Clearly oracle has no interest in maintaining these names. Let's get them handed over to more suitable custodians

**Comment by graingert** [comment]
Clearly oracle has no interest in maintaining these names. Let's get them handed over to more suitable custodians

**Comment by geertjanvdk** [comment]
No?

**Comment by graingert** [comment]
I only brought up mysql-connector-python as it's another unmaintained package that can be transferred to this team

**Comment by geertjanvdk** [comment]
I don't quite get it.. MySQL Connector/Python is maintained and available from www.mysql.com. Why and how would you transfer a package or even a name? How did you decide it was not maintained?

**Comment by graingert** [comment]
Sure there's something else maintained under the same name away from PyPI. but Oracle never uploaded anything to PyPI. It should be moved to a team that will use the name. Just like MySQL-Python

**Comment by geertjanvdk** [comment]
Where is the rule written that it should be moved to somebody else? I would need a link to PyPI where this is mentioned, so I can pull strings.

**Comment by graingert** [comment]
It would be much better to have an official policy like npm, but there
isn't one for PyPI. Richard just transfers package names if the name is
unused in response to a support request.

**Comment by geertjanvdk** [comment]
@graingert This is amazing. So if I wouldn't have commented here, it could have been that 'mysql-connector-python' would be transferred to who-ever? No, I can not believe that.
Again, we want to upload, but we can not, and ball is in PyPI's camp.
I'll try to email again to see if something moved. Would be great to get something downloadable (would help me personally as well).

**Comment by graingert** [comment]
Use it or lose it

**Comment by graingert** [comment]
Also there's nothing stopping me uploading the connector under another name? Did you guys just forget the password to the account?

**Comment by zzzeek** [comment]
@graingert why do you care about the "mysql-connector-python" name ?   That's Oracle's driver, they have a placeholder page on pypi for it that points people who are interested towards its own download infrastructure.  Given that Oracle doesn't want to put the source on pypi for downloading renders the project as not very useful and sadly in a fairly marginalized state, but as there are other drivers that remain very popular, I don't see the problem - the community just uses mysql-python  / mysqlclient / pymysql these days.

**Comment by geertjanvdk** [comment]
@zzzeek agreed, except that we never said "we don't want"..

**Comment by graingert** [comment]
@zzzeek it's PyPI that won't let oracle upload their package

**Comment by zzzeek** [comment]
@geertjanvdk "want' is kind of a word we can parse into :)  I'm not involved w/ pypi but there's AFAIK no technical reason prohibiting Oracle from putting up the code.   If the reason is strictly technical, I'm sure we can work out with @dstufft how to work that out.   It's hard for us on the outside to talk about this issue since as you just mentioned a few messages upwards, you "should not discuss this" and the blog post you refer to doesn't give any reasons either, so we can only assume there's some corporate / legal reason preventing this from happening.  Seems like IBM DB2 doesn't have their driver up either.   shrugs

**Comment by zzzeek** [comment]
this is the first I'm hearing that Pypi won't allow it.  By what mechanism are they "disallowing" it?   name is explcitly blocked?  IP address?  parsing the tar.gz file for some magic file ?

**Comment by graingert** [comment]
@zzzeeek
> the PyPI license gives issues when uploading GPL'ed software.

**Comment by dstufft** [comment]
Uhhh, what? There's like basically zero restrictions to uploading to PyPI except:

* You have to grant us a license to distribute whatever you upload (and if the thing is GPL, that's already a part of the license). It's like the bare minimum required to make PyPI legally able to serve the files that people upload...
* Some basic file checks for things like "you can't upload a JPEG and call it a Python package".

**Comment by geertjanvdk** [comment]
@zzzeek it's not technical, nothing to do with the actual uploading through web.. I can not just go into details, indeed.. :/ But this is IMHO out-of-scope.. It's a problem we have to deal with.

OK, lets wrap up.

I will not agree that 'mysql-connector-python' is transferred for no good reason to an unknown 3rd party. There is also "NO" reason to do so.

The MySQL and Python Community has plenty of choice of what the use. Yes, PyPI is a popular way of getting Python packages, but it is not the only way. 'mysql-connector-python' in PyPI stays where it is until we can figure it out. One can always open a request at bugs.mysql.com to check up on it.

EOF

**Comment by graingert** [comment]
@dstufft

> Uhhh, what?


Oracle claim to have already contacted you, and that the "ball is in (your) court"

**Comment by edmorley** [comment]
> One can always open a request at bugs.mysql.com to check up on it.

Bugs are already on file for this, but haven't had any updates.

Bug 76063: pypi package requires deprecated external hosting (opened Feb 2015):
http://bugs.mysql.com/bug.php?id=76063

Bug 78982: Update the pypi version of the connector (opened Oct 2015)
http://bugs.mysql.com/bug.php?id=78982

**Comment by graingert** [comment]
@dstufft can we please have an official policy to prevent this sort of abuse/squating like npm have?

**Comment by dstufft** [comment]
*Puts PyPI Administrator Hat on*

We are not going to transfer ``mysql-connector-python`` or any other actually used name to someone without the permission of someone with rights on PyPI except in cases like: Every one who can release Django to PyPI has disappeared, but we can tell that the project itself still exists and is controlled by the DSF. The only other situation we'd do anything like this is cases where there is a legal requirement to transfer a name OR if a name violates our CoC we "block" the name from use by transferring it to ourselves. We haven't given PIL to Pillow and we didn't give setuptools to distribute. Names are first come first serve and the most we generally do is facilitate communication between whoever currently owns the name and whoever wants it.

This is different for names that are *not* used but are registered (and by not used, I mean, no information is available on PyPI, no releases, no URLs, nothing except an entry in the database saying "X owns Y" and even then we only do that after contacting the original owner and asking them if it's OK. If they respond that they want to keep it then we don't transfer it.

So basically we only ever transfer if the current owner's agree to it OR if the project has zero information and is obviously a placeholder AND the original owner is non responsive after some number of months.

I have no idea why Oracle believes it cannot upload to PyPI, nobody has said anything to me about it (unless I have forgotten...). It's possible they have contacted Richard instead though, but if so he's said nothing to me and is largely emeritus now. They also might have contacted the PSF itself rather than me, which is fine as well (particularly if it's a legal issue since Van is the guy who deals with that).

If there's something I'm supposed to be doing to help Oracle upload, then I was either unaware or I forgot-- but it's possible (and even likely!) that they don't mean me, but either Richard or the PSF.

**Comment by graingert** [comment]
This falls under:

> Every one who can release Django to PyPI has disappeared,

Everyone who can release mysql-connector-python has disappeared or perhaps never existed.

**Comment by geertjanvdk** [comment]
@dstufft Ok, I agree with what you posted there about PyPI. Sounds good. There will be a transfer of ownership soon for that entry anyway.

The other things are bug reported.. and are still nagging yes. @edmorley pointed that correctly out.

@graingert So, nothing changes.

Thanks late night thread :)

**Comment by graingert** [comment]
@geertjanvdk everyone who can upload mysql-connector-python has disappeared or never existed, thus under policy the name should be transferred to someone who can make the release

**Comment by dstufft** [comment]
@graingert Well, one of the user's of mysql-connector-python is "GeertVanderkelen" which has the same avatar and real life name as @geertjanvdk, so I'm pretty sure he is an owner on PyPI. If you have reason to believe they're not the same I can send an email to the address on file instead, but I assume it'll just be a waste of time, unless there are two people named "Geert Vanderkelen" who work on mysql-connector-python (or one is an imposter). Otherwise, @geertjanvdk has already indicated in this thread they do not want the name transferred.

**Comment by JimAtOracle** [comment]
It seems like this may have been halfway resolved in any event, but I just thought I'd weigh in here re: the root cause of the problem (esp. since it sounds like Donald was not made aware of the issue).  

This is Jim Wright, Oracle's Chief Architect for Open Source Policy, Strategy, Compliance and Alliances.  I am an attorney for Oracle as well as an architect and I'm not sure if you folks are represented in this matter - if you are, the Rules of Professional Conduct for attorneys are such that I cannot communicate with you without your counsel's consent, so please have them email me at jim.wright@oracle.com.  (And if you're not represented, please email me yourself to that effect if you'd like to discuss.)

The terms of use at https://www.python.org/about/legal/ provide in pertinent part as follows:

"third party content providers grant the PSF and all other users of the web site an irrevocable, worldwide, royalty-free, nonexclusive license to reproduce, distribute, transmit, display, perform, and publish such content, including in digital form"

Unfortunately this does not say the PSF and its users are getting a license under the GPL.  In fact, the terms of that license grant are clearly incompatible with the GPL (e.g., the GPL is not an irrevocable license - it contains an express termination provision that is invoked if you do not comply with its terms).  If content providers are bound by those terms to begin with, for someone to upload GPLed content under that license probably either means they are violating the GPL (e.g., if someone else uploaded Oracle's GPL licensed content), or, if they are the sole copyright holder, they are granting the PSF and all its users a *different* license than the GPL to that material.  That doesn't work for the obvious reasons. 

If I'm informed correctly, this issue with the terms of use was raised with the PSF some time ago but it sounds like this may have gotten lost in the noise - if you'd like to discuss how to fix it, I would love to help in whatever way I can, and as soon as the terms are fixed, we would be happy to maintain MySQL Connector/Python on PyPI.

  -- Jim

**Comment by graingert** [comment]
@geertjanvdk just indicated that he can't make a release, isn't this grounds for the package to be transferred?

**Comment by geertjanvdk** [comment]
I'm happy about two things: 1) I am not disappeared; and 2) the "I can't" was explained by Jim; I'll be sleeping well tonight.

**Comment by graingert** [comment]
Although @JimAtOracle you do own copyright to mysql-connector-python, so you could assign such a licence

**Comment by graingert** [comment]
(if you so wished)

**Comment by graingert** [comment]
@geertjanvdk I'm hearing a "won't" not a "can't", which I guess complies with PyPI policy

**Comment by zzzeek** [comment]
did @jimatoracle just delete his message?   come on, guys

**Comment by dstufft** [comment]
Hmm, it appears that @JimAtOracle deleted their comment, with respect to that I won't quote it here. I am not a Lawyer and I don't make decisions about the legal side of PyPI, I just run the site itself :)

The current ToS was crafted by @VanL and I can only assume that he is of the opinion it doesn't clash, since I'm pretty sure he does not want to prevent GPL software from being uploaded to PyPI. However, this is a sufficiently nuanced discussion that I suspect Github issues is not the correct place to hash out the nuances of law here. If Oracle has not yet already, please reach out to the PSF board (psf@python.org) and indicate why you believe you can't upload to PyPI. The board is better equipped to handle this than I suspect I, and a bunch of other folks in a GitHub issue are. If you want you can CC me (donald@python.org) or you can just reach out to them directly.

**Comment by graingert** [comment]
> It seems like this may have been halfway resolved in any event, but I just thought I'd weigh in here re: the root cause of the problem (esp. since it sounds like Donald was not made aware of the issue).

This is Jim Wright, Oracle's Chief Architect for Open Source Policy, Strategy, Compliance and Alliances. I am an attorney for Oracle as well as an architect and I'm not sure if you folks are represented in this matter - if you are, the Rules of Professional Conduct for attorneys are such that I cannot communicate with you without your counsel's consent, so please have them email me at jim.wright@oracle.com. (And if you're not represented, please email me yourself to that effect if you'd like to discuss.)

The terms of use at https://www.python.org/about/legal/ provide in pertinent part as follows:

"third party content providers grant the PSF and all other users of the web site an irrevocable, worldwide, royalty-free, nonexclusive license to reproduce, distribute, transmit, display, perform, and publish such content, including in digital form"

Unfortunately this does not say the PSF and its users are getting a license under the GPL. In fact, the terms of that license grant are clearly incompatible with the GPL (e.g., the GPL is not an irrevocable license - it contains an express termination provision that is invoked if you do not comply with its terms). If content providers are bound by those terms to begin with, for someone to upload GPLed content under that license probably either means they are violating the GPL (e.g., if someone else uploaded Oracle's GPL licensed content), or, if they are the sole copyright holder, they are granting the PSF and all its users a different license than the GPL to that material. That doesn't work for the obvious reasons.

If I'm informed correctly, this issue with the terms of use was raised with the PSF some time ago but it sounds like this may have gotten lost in the noise - if you'd like to discuss how to fix it, I would love to help in whatever way I can, and as soon as the terms are fixed, we would be happy to maintain MySQL Connector/Python on PyPI.

-- Jim

**Comment by zzzeek** [comment]
@graingert now oracle is going to send you a dmca notice

**Comment by graingert** [comment]
me? But I'm nothing to do with github

**Comment by geertjanvdk** [comment]
Guys, Jim messaged me something went wrong with his GitHub account, but "comment will no doubt reappear".

**Comment by graingert** [comment]
Is deleting a comment you were not allowed to send tampering with evidence?

**Comment by JimAtOracle** [comment]
Ok, I think my account has been reactivated... pretty funny to see the conjecture/responses though.

I have contacted the PSF board as per Donald's suggestion and hopefully we can get the terms questions ironed out.  Cheers all.

**Comment by dstufft** [comment]
To be clear, Van is not my lawyer, nor do I have a lawyer at all, at least to my knowledge. I also don't really have a horse in this race :) I don't care what Terms PyPI has as long as they allow us to achieve our mission nor do I use MySQL Connector. I mostly just develop and operate PyPI on a technical level as a volunteer, on a legal level it's all the PSF.

That being said, I don't think it's very useful to discuss it with me :) I am not a lawyer, Van once tried to explain how the PyPI ToS interacts with the license already on software to me and I got confused very quickly. Although I recall a previous time when Van went into a bit of detail about the license when there was a similar question about the terms of the license. I don't really understand what Oracle's concern is so I don't know that it's the same, or even really related, but in case it's interesting you can view the thread archives at https://mail.python.org/pipermail/python-legal-sig/2013-March/thread.html.

Anyways, it sounds like things are in good hands, and hopefully Jim and the PSF can get things hashed out to the point everyone is happy and there's nothing stopping Oracle from uploading to PyPI.

**Comment by methane** [comment]
tl;dr
My understands are:

* Oracle doesn't concern about "MySQL-python" package name at PyPI.
* Oracle has something to worry about uploading GPL package to PyPI.
  * PyPI terms of use must permit uploading GPL packages.  "pip installable" makes users happy.

**Comment by methane** [comment]
I'm sorry about I misunderstand like "Both of Oracle and I claim taking over MySQL-python on PyPI."

@graingert Are you PyPI maintainer?
I doesn't understand why this discussion is happen here.

**Comment by graingert** [comment]
@methane I wanted to sort out the other python mysql DB API library as a sort of rider on this issue. With the aim to get them all under one umbrella in one go

**Comment by zzzeek** [comment]
@graingert im not sure if i understand that statement.  are you saying you want all MySQL DBAPI libraries to be maintained by one group ?   how would that even be achievable, if someone decides to write a new DBAPI for MySQL...

**Comment by tiwilliam** [comment]
Can we please open a separate issue for discussing `mysql-connector-python`? This issue was discussing how to get PyMySQL imported under its original name, either by changing the module name or finding the PyPi owner of `MySQL-python`.

**Comment by graingert** [comment]
No, only the unmaintained ones

**Comment by methane** [comment]
@graingert MySQL Connector/Python is maintained project.
In fact, it's actively developed, while this project is mostly maintenance only mode.

Could you go
1) bugs.mysql.com about MySQL Connector/Python
2) distutils-sig ML about general discussion of taking over unmaintained projects on PyPI
3) https://sourceforge.net/p/pypi/support-requests/553/ about taking over MySQL-python on PyPI

**Comment by methane** [comment]
This issue is too long.  I can't read whole thread already.
I lock this until https://sourceforge.net/p/pypi/support-requests/553/ is resolved.

**Comment by methane** [comment]
PEP 541 is accepted.  But this issue is too long and old.
Continue discussion on new issue #232.
